### PR TITLE
Include info icon with state notes

### LINKED ISSUE #1041: Add info icon and tooltip next to statename in table IF a note needs to be shown
**Is your feature request related to a problem? Please describe.**
When a remark needs to be made regarding a state's number in the main table, it's currently added to the "About (FAQ)" page. Instead of that, it should be a ℹ️ icon, which on tap/hover pops up a tool-tip.

**Describe the solution you'd like**
1. Fetch the `statenotes` field now available in the `statewise` object for each state at [https://api.covid19india.org/data.json ](https://api.covid19india.org/data.json)
image

2. Show an info icon ℹ️ next to the statenames IF there is a valid non-empty string for `statenotes`
3. Show the text as tooltip

**Describe alternatives you've considered**
Currently we show the info message in FAQ for WB.
This is not very useful, now that we have lot more caveats to display.

**Additional context**
1. Data volunteers can update the Statewise sheet to update the notes.
2. Maybe an empty string check might be useful, to avoid just a space in the `statenotes` to cause a ℹ️ to show up.

### PR DESCRIPTION
**Pull request by ameya-pandilwar**
**Description of PR**
Include info icon with state notes

**Type of PR**

- [ ] Bugfix
- [x] New feature

**Relevant Issues**  
Fixes #1041 

**Checklist**

- [x] Compiles and passes lint tests
- [x] Properly formatted
- [x] Tested on desktop
- [x] Tested on phone

**Screenshots**
Before
Screen Shot 2020-04-13 at 10 18 52 PM

After
Screen Shot 2020-04-13 at 10 18 10 PM

### CODE REVIEW

**Comment by sudevschiz** [comment]
@nayakrahul Thanks for working on this. Can you make few changes : 


1. It's tough to tap on the info icon on phones because that could cause district drop-down. Do you have any better ideas to improve UX? Just make the info icon larger to tap?
2. Possible to make the dialog bubble a bit transparent so that other numbers are not hidden fully?
3. There are some leading white-spaces. Possible to remove those?

**Comment by nayakrahul** [comment]
> @nayakrahul Thanks for working on this. Can you make few changes :
> 
> 1. It's tough to tap on the info icon on phones because that could cause district drop-down. Do you have any better ideas to improve UX? Just make the info icon larger to tap?
> 2. Possible to make the dialog bubble a bit transparent so that other numbers are not hidden fully?
> 3. There are some leading white-spaces. Possible to remove those?

On mobile devices, when we tap on info icon, we should only show tooltip but not invoke the district dropdown. Is this possible? @sudevschiz

**Comment by ameya-pandilwar** [comment]
@sudevschiz -- Enhanced the following --

1. Increased the icon size to make it easier to tap.
2. Made the background a little transparent.
3. Aligned the text appropriately.

Browser --
Screen Shot 2020-04-14 at 9 19 21 PM

Mobile --
Screen Shot 2020-04-14 at 9 19 45 PM

Please review and merge!

**Comment by ameya-pandilwar** [comment]
@sudevschiz Please review and merge!

**Comment by tannakartikey** [comment]
@sudevschiz @navneetgupta on mobile it still opens the district dropdown. Not sure if it is intended or need changes.

@navneetgupta please rebase the PR to remove conflicts.

**Comment by ameya-pandilwar** [comment]
@tannakartikey -- Rebased to remove conflicts.

File: src/components/row.js [hunk 1]
```diff
@@ -334,4 +345,20 @@ function Row(props) {
   );
 }
 
+const tooltipStyle = {
```
Comment (tannakartikey) [inline src/components/row.js]: @ameya-pandilwar can you move the style to CSS file? The pattern you followed by storing it to a var is not used anywhere else in the file.

**Review by tannakartikey — CHANGES_REQUESTED** [review]

**Review by tannakartikey — CHANGES_REQUESTED** [review]
The tooltip works fine for the first time. If you click on the different tooltip. But if we click the same tooltip continuously to open/close it does not work nicely. (attached gif)
It is working fine on the desktop.
ezgif com-crop

File: src/components/row.js (hunk 1 above)
Comment (ameya-pandilwar) [reply src/components/row.js]: @tannakartikey sure, will change it to an inline style to make it consistent with the existing style

**Comment by ameya-pandilwar** [comment]
> The tooltip works fine for the first time. If you click on the different tooltip. But if we click the same tooltip continuously to open/close it does not work nicely. (attached gif)
> It is working fine on the desktop.
> ezgif com-crop

I'm using the `react-lightweight-tooltip`. @tannakartikey Could you suggest a way to fix this behavior?

**Comment by tannakartikey** [comment]
@ameya-pandilwar the bug is in the library itself. 
The library is not maintained since 2 years, however, I see that there are lots of [PRs](https://github.com/mcumpl/react-lightweight-tooltip/pulls) created for the library.

I suggest that we use the different properly maintained library or use the patch _if_ it is available in those PRs.

If the above does not work then either we handcraft the tooltip or hide it on the mobile.

**Review by tannakartikey — APPROVED** [review]
@ameya-pandilwar approving the PR as we want to show this information to users and the bugs are not critical.

If you come up with a better solution please feel free to create another PR.

Thanks for your contribution!

**Pull request merged**